crates/neon/src/object/mod.rs

```diff
@@ -134,8 +143,90 @@ impl<'a> PropertyKey for &'a str {
     }
 }
 
+/// A builder for accessing an object property.
+///
+/// The builder methods make it convenient to get and set properties
+/// as well as to bind and call methods.
+/// ```
+/// # use neon::prelude::*;
+/// # fn foo(mut cx: FunctionContext) -> JsResult<JsString> {
+/// # let obj: Handle<JsObject> = cx.argument(0)?;
+/// let x: f64 = obj
+///     .prop(&mut cx, "x")
+///     .get()?;
+///
+/// obj.prop(&mut cx, "y")
+///     .set(x)?;
+///
+/// let s: String = obj.prop(&mut cx, "toString")
+///     .bind()?
+///     .apply()?;
+/// # Ok(cx.string(s))
+/// # }
+/// ```
+pub struct PropOptions<'a, 'cx, K>
+where
+    'cx: 'a,
+    K: PropertyKey,
+{
+    pub(crate) cx: &'a mut Cx<'cx>,
+    pub(crate) this: Handle<'cx, JsObject>,
+    pub(crate) key: K,
+}
+
+impl<'a, 'cx, K> PropOptions<'a, 'cx, K>
+where
+    'cx: 'a,
+    K: PropertyKey,
+{
+    /// Gets the property from the object and attempts to convert it to a Rust value.
+    /// Equivalent to calling `R::from_js(cx, obj.get(cx)?)`.
+    ///
+    /// May throw an exception either during accessing the property or converting the
+    /// result type.
+    pub fn get<R: TryFromJs<'cx>>(&mut self) -> NeonResult<R> {
+        let v = self.this.get_value(self.cx, self.key)?;
+        R::from_js(self.cx, v)
+    }
+
+    /// Sets the property on the object to a value converted from Rust.
+    /// Equivalent to calling `obj.set(cx, v.try_into_js(cx)?)`.
+    ///
+    /// May throw an exception either during converting the value or setting the property.
+    pub fn set<V: TryIntoJs<'cx>>(&mut self, v: V) -> NeonResult<bool> {
+        let v = v.try_into_js(self.cx)?;
+        self.this.set(self.cx, self.key, v)
+    }
+
+    /// Gets the property from the object as a method and binds `this` to the object.
+    ///
+    /// May throw an exception either during accessing the property or downcasting it
+    /// to a function.
+    pub fn bind(&'a mut self) -> NeonResult<BindOptions<'a, 'cx>> {
+        let callee: Handle<JsValue> = self.this.get(self.cx, self.key)?;
+        let this = Some(self.this.upcast());
+        Ok(BindOptions {
+            cx: self.cx,
+            callee,
+            this,
+            args: smallvec![],
+        })
+    }
+}
+
 /// The trait of all object types.
 pub trait Object: Value {
```

Comment:
Let's make them `#[doc(hidden)]` for now and we can file a followup issue to formally deprecate them a minor version or two down the road.

Reply:
I made the following methods `#[doc(hidden)]`:
- `get_opt()`
- `get()`
- `get_value()`
- `set()`
- `call_method_with()`

And I've replaced all the instances of them in the doc comments.

Next I'll do the same for `.call()`, `.construct()`, `.call_with()`, `.construct_with()`, once I've implemented the full range of replacements via `.bind()`.

Reply:
I found a better mechanism than `#[doc(hidden)]`: Rust now allows for soft deprecation using `#[deprecated(since = "TBD")]`. This will leave a deprecation note in the API docs ([example](https://deploy-preview-1056--neon-bindings-docs.netlify.app/neon/object/trait.object#method.get)) but not generate compiler warnings. Much better!

Reply:
Replaced all of `.call()`, `.construct()`, `.call_with()`, and `.construct_with()` with `.bind()` in the doc comments.